File: WebApp/Models/FormTemplate.cs
using LightMethods.Survey.Models.Entities;
using LightMethods.Survey.Models.MetricFilters;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace WebApi.Models
{
    public class FormTemplateDTO
    {

        public Guid Id { set; get; }

        public FormTemplateCategoryDTO FormTemplateCategory { set; get; }

        public Guid? ProjectId { set; get; }

        public Guid CreatedById { set; get; }

        public string ProjectName { get; set; }

        public string Code { get; set; }

        public string Title { get; set; }

        public double Version { get; set; }

        public string Description { set; get; }

        public bool IsPublished { set; get; }

        public string Colour { set; get; }
        public string DescriptionFormat { get; set; }

        public Guid? CalendarDateMetricId { set; get; }

        public Guid? TimelineBarMetricId { set; get; }

        public ICollection<MetricGroupDTO> MetricGroups { set; get; }

        public List<MetricFilter> MetricFilters { get; set; }

        public FormTemplateDTO()
        {
            this.MetricFilters = new List<MetricFilter>();
        }
    }
}

File: WebApp/wwwroot/scripts/domain/formTemplate.ts
module App.Models {
    "use strict";

    export enum FormTemplateDiscriminators {
        regularThread = 0,
        adviceThread = 1
    };

    export interface IFormTemplate extends angular.resource.IResource<IFormTemplate> {
        id: string;
        projectId: string;
        organisation: IOrganisation;
        code: string;
        title: string;
        project: IProject;
        projectName: string;
        description: string;
        colour: string;
        formTemplateCategory: IFormTemplateCategory;
        discriminator: FormTemplateDiscriminators;
        metricGroups: IMetricGroup[];
        isChecked: boolean;
        calendarDateMetricId: string;
        descriptionFormat: string;
        timelineBarMetricId: string;
        canView?: boolean;
        canAdd?: boolean;
        canEdit?: boolean;
        canDelete?: boolean;
    }

    export interface IThreadAssignment {
        formTemplateId: string;
        orgUserId: string;
        orgUserName: string;
        accountType: number;
        email: string;
        isRootUser: boolean;
        canAdd: boolean;
        canEdit: boolean;
        canView: boolean;
        canDelete: boolean;
    }
}